pvlib/iotools/era5.py

```diff
@@ -0,0 +1,294 @@
+    """
+    Retrieve ERA5 reanalysis data from the Copernicus Data Store (CDS).
+
+    * Temporal coverage: 1979 to present (latency of ~5 days)
+    * Temporal resolution: hourly
+    * Spatial coverage: global
+    * Spatial resolution: 0.25° by 0.25°
+
+    An overview of ERA5 is given in [1]_ and [2]_. Data is retrieved using the
+    CDSAPI [3]_.
+
+    .. admonition:: Time reference
+
+        ERA5 time stamps are in UTC and corresponds to the end of the period
+        (right labeled). E.g., the time stamp 12:00 for hourly data refers to
+        the period from 11.00 to 12:00.
+
+    .. admonition:: Usage notes
+
+        To use this function the package CDSAPI [4]_ needs to be installed
+        [3]_. The CDSAPI keywords are described in [5]_.
+
+        Requested variables should be specified according to the naming
+        convention used by the CDS. The returned data contains the short-name
+        versions of the variables. See [2]_ for a list of variables names and
+        units.
+
+        Access to the CDS requires user registration, see [6]_. The obtaining
+        API key can either be passed directly to the function or be saved in a
+        local file as described in [3]_.
+
+        It is possible to check your
+        `request status <https://cds.climate.copernicus.eu/cdsapp#!/yourrequests>`_
+        and the `status of all queued requests <https://cds.climate.copernicus.eu/live/queue>`_.
+
+    Parameters
+    ----------
+    latitude: float or list
+        in decimal degrees, between -90 and 90, north is positive (ISO 19115).
+        If latitude is a list, it should have the format [S, N] and
+        latitudes within the range are selected according to the grid.
+    longitude: float or list
+        in decimal degrees, between -180 and 180, east is positive (ISO 19115).
+        If longitude is a list, it should have the format [W, E] and
+        longitudes within the range are selected according to the grid.
+    start: datetime like
+        First day of the requested period
+    end: datetime like
+        Last day of the requested period
+    api_key: str, optional
+        Personal API key for the CDS
+    variables: list, default: ERA5_DEFAULT_VARIABLES
+        List of variables to retrieve (according to CDS naming convention)
+    dataset: str, default 'reanalysis-era5-single-levels'
+        Name of the dataset to retrieve the variables from. Can be either
+        'reanalysis-era5-single-levels' or 'reanalysis-era5-land'.
+    product_type: str, {'reanalysis', 'ensemble_members', 'ensemble_mean',
+                        'ensemble_spread'}, default: 'reanalysis'
+        ERA5 product type
+    grid: list or tuple, default: (0.25, 0.25)
+        User specified grid resolution
+    save_path: str or path-like, optional
+        Filename of where to save data. Should have ".nc" extension.
+    cds_client: CDS API client object, optional
+        CDS API client
```

Comment:
What's the use case for having this be a parameter?

Reply:
Probably none... I'd seen other wrappers do it. Maybe if you wanted to create your own API client and submit multiple jobs? I'm ok removing it.